src/NHapi.Base/Parser/MessageIterator.cs

```diff
@@ -40,12 +41,18 @@ static MessageIterator()
         }
 
         /// <summary>Creates a new instance of MessageIterator. </summary>
-        public MessageIterator(IMessage start, IStructureDefinition startDefinition, string direction, bool handleUnexpectedSegments)
+        public MessageIterator(
+            IMessage start,
+            IStructureDefinition startDefinition,
+            string direction,
+            bool handleUnexpectedSegments,
+            ParserBase parser)
```

Comment:
I'm not sure we need to pass the entire parser in here perhaps just the `ParserConfiguration`?

Reply:
If we only pass the `ParserConfiguration` then we lose the ability to distinguish the type of parser being used. However it seems that this `MessageIterator` is only ever used by `PipeParser` so that is not an issue. I will update this to take `ParserConfiguration`.

As an aside, it also looks like there are two versions of `MessageIterator` in the `NHapi.Base` project:
* `NHapi.Base.Parser`
* `NHapi.Base.Util`

Reply:
The `MessageIterator` in `NHapi.Base.Util` is the legacy version, left in the project for use by the `LegacyPipeParser`, which was the `PipeParser` before v3 of nhapi

Is the hapi version of the message iterator used for both the `PipeParser` and the `XmlParser`?

Reply:
Looking at the Java source code no it isnt.